components/telemetry-operator/internal/fluentbit/config/builder/namespace_filter.go

```diff
@@ -0,0 +1,39 @@
+package builder
+
+import (
+	"fmt"
+	"strings"
+
+	telemetryv1alpha1 "github.com/kyma-project/kyma/components/telemetry-operator/apis/telemetry/v1alpha1"
+)
+
+func systemNamespaces() []string {
+	return []string{"kyma-system", "kyma-integration", "kube-system", "istio-system", "compass-system"}
+}
+
+func createNamespaceGrepFilter(pipeline *telemetryv1alpha1.LogPipeline) string {
+	namespaces := pipeline.Spec.Input.Application.Namespaces
+	if namespaces.System {
+		return ""
+	}
+
+	var sectionBuilder = NewFilterSectionBuilder().
+		AddConfigParam("Name", "grep").
+		AddConfigParam("Match", fmt.Sprintf("%s.*", pipeline.Name))
+
+	if len(namespaces.Include) > 0 {
+		return sectionBuilder.
+			AddConfigParam("Regex", fmt.Sprintf("$kubernetes['namespace_name'] %s", strings.Join(namespaces.Include, "|"))).
```

Comment:
Is there a reason why we build the rule differently for the rewrite_tag filter and regex filter?

`"^(namespace1|namespace2)$"` vs `namespace1|namespace2`

Reply:
Not really, just figured since grep filter has explicit config params for include (`regex`) and `exclude` the regex kann be simpler and doesn't need capturing group etc.
Probably rewrite_tag regex for containers could also be simplified but didn't test that out.
Good catch, we should have a look at the rewrite_tag regex creation as well and see if it can be simplified to save a few steps while matching